storages/jsonl-storage/src/store_mut.rs

```diff
@@ -0,0 +1,178 @@
+use {
+    crate::{
+        error::{JsonlStorageError, OptionExt, ResultExt},
+        JsonlStorage,
+    },
+    async_trait::async_trait,
+    gluesql_core::{data::Schema, prelude::Key, result::Result, store::DataRow, store::StoreMut},
+    serde_json::{Map, Value as JsonValue},
+    std::{
+        fs::{remove_file, File, OpenOptions},
+        io::Write,
+        {cmp::Ordering, iter::Peekable, vec::IntoIter},
+    },
+};
+
+#[async_trait(?Send)]
+impl StoreMut for JsonlStorage {
+    async fn insert_schema(&mut self, schema: &Schema) -> Result<()> {
+        let data_path = self.data_path(schema.table_name.as_str());
+        File::create(data_path).map_storage_err()?;
+
+        if schema.column_defs.is_some() {
+            let schema_path = self.schema_path(schema.table_name.as_str());
+            let ddl = schema.to_ddl();
+            let mut file = File::create(schema_path).map_storage_err()?;
+            write!(file, "{ddl}").map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn delete_schema(&mut self, table_name: &str) -> Result<()> {
+        let data_path = self.data_path(table_name);
+        if data_path.exists() {
+            remove_file(data_path).map_storage_err()?;
+        }
+
+        let schema_path = self.schema_path(table_name);
+        if schema_path.exists() {
+            remove_file(schema_path).map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn append_data(&mut self, table_name: &str, rows: Vec<DataRow>) -> Result<()> {
+        let schema = self
+            .fetch_schema(table_name)?
+            .map_storage_err(JsonlStorageError::TableDoesNotExist)?;
+        let table_path = self.data_path(table_name);
+
+        let mut file = OpenOptions::new()
+            .write(true)
+            .append(true)
+            .open(table_path)
+            .map_storage_err()?;
+
+        let column_defs = schema.column_defs.unwrap_or_default();
+        let labels = column_defs
+            .iter()
+            .map(|column_def| column_def.name.as_str())
+            .collect::<Vec<_>>();
+
+        for row in rows {
+            let json_string = match row {
+                DataRow::Vec(values) => {
+                    let mut json_map = Map::new();
+                    for (key, value) in labels.iter().zip(values.into_iter()) {
+                        json_map.insert(key.to_string(), value.try_into()?);
+                    }
+
+                    JsonValue::Object(json_map).to_string()
+                }
+                DataRow::Map(hash_map) => {
+                    let mut json_map = Map::new();
+                    for (key, value) in hash_map {
+                        json_map.insert(key.to_string(), value.try_into()?);
+                    }
+
+                    JsonValue::Object(json_map).to_string()
+                }
+            };
+            writeln!(file, "{json_string}").map_storage_err()?;
+        }
```

Comment:
nit. this would be more simple.
```rust
for row in rows {
    let json_string = match row {
        DataRow::Vec(values) => labels
            .iter()
            .zip(values.into_iter())
            .map(|(key, value)| Ok((key.to_string(), value.try_into()?)))
            .collect::<Result<Map<String, JsonValue>>>(),
        DataRow::Map(hash_map) => hash_map
            .into_iter()
            .map(|(key, value)| Ok((key.to_string(), value.try_into()?)))
            .collect(),
    }
    .map(JsonValue::Object)?
    .to_string();

    writeln!(file, "{json_string}").map_storage_err()?;
}
```